### PR TITLE
Add VCR

### LINKED ISSUE #282: Add VCR for specs
More info here: https://benoittgt.github.io/vcr/

I had some experience with it, so I could set it up after merging my existing pull requests.

Main benefits:
* no more `BOT_API_TOKEN` is required for specs (currently we even don't have instructions for it in README);
* we're not testing third-party server (Telegram) availability (we'll able to run specs even in outages);
* speed of specs, obviously.

Cons:
* new files in git repository (with masked secrets, like tokens);
* we have to watch for Telegram API updates manually and update our cassettes (or have some TTL for them, but regularly it's not required).

### PR DESCRIPTION
**Pull request by AlexWayfer**
Write all requests to cassetes.
* Speed up specs.
* Don't make new requests for every run of existing specs.

Resolve #282.

### CODE REVIEW

File: spec/cassettes/Telegram_Bot_Api/_call/has_result.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      access-control-allow-origin:
+      - "*"
+      access-control-allow-methods:
+      - GET, POST, OPTIONS
+      access-control-expose-headers:
+      - Content-Length,Content-Type,Date,Server,Connection
+    body:
+      encoding: UTF-8
+      string: '{"ok":true,"result":{"id":2200529744,"is_bot":true,"first_name":"Test
+        bot","username":"alexwayfer_test_bot","can_join_groups":true,"can_read_all_group_messages":false,"supports_inline_queries":false}}'
```
Comment (AlexWayfer) [inline spec/cassettes/Telegram_Bot_Api/_call/has_result.yml:40]: I can mask some data here if it confuses you, but I see nothing confidential here, and see no problems with different data for different cassettes (if somebody else will write new specs or re-record existing ones).

**Comment by AlexWayfer** [comment]
I've reproduced the problem from CI: `BOT_API_TOKEN=` (empty value).

I'll fix and make the PR ready.

File: spec/cassettes/Telegram_Bot_Api/_call/has_result.yml (hunk 1 above)
Comment (atipugin) [reply spec/cassettes/Telegram_Bot_Api/_call/has_result.yml:40]: Well yeah, nothing sensitive here. Let's leave it like this for now.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,4 +1,21 @@
 # frozen_string_literal: true
 
 require 'dotenv/load'
+
+ENV['BOT_API_ENV'] ||= 'test'
```
[the remark this answers is not in the document]
Comment (AlexWayfer) [reply spec/spec_helper.rb:5]: The value doesn't mean anything. And it's not required for VCR specs. So… anyone can clone the repo, run specs and get success, without additional actions like receiving a token.

But if you want to write new specs, or rewrite existing cassettes — you should set your token and only then they'll execute (once) and record.

Exactly here, as you can see from my comments above, without any value at all we'll get exceptions. So I see no point in removing these defaults.

**Review by atipugin — APPROVED** [review]

**Pull request merged**